cmake/toolchain/llvm/Kconfig

```diff
@@ -17,3 +17,15 @@ config LLVM_USE_LLD
 	  Use LLVM built-in lld linker with llvm/clang.
 
 endchoice
+
+config LLVM_PROFILE
```

Comment:
Should this really be named `LLVM_PROFILE` ?

I acknowledge it's different than our current `COVERAGE` feature, but it is still desirable that we try to keep names more general so that the Kconfig name can cover more that just a single toolchain, and so far, the `-fprofile-instr-generate -fcoverage-mapping` is still coverage, just a different variant / output.

I don't have a proposal atm, but would like to prevent situations where each toolchain has their own setting names for features that should be general (although their exact implementation / flags may differ).

Reply:
Is there no way to determine if the toolchain is LLVM or gcc-compatible or otherwise?

It seems like this would be trivial to do (at a low level) if that were the case. I guess getting  things in similar reporting formats is more of a twister problem

Reply:
What about a choice dependent on COVERAGE, of COVERAGE_NATIVE_SOURCE or COVERAGE_NATIVE_GCOV or COVERAGE_GCOV.
where:
* COVERAGE_NATIVE_GCOV would be the default with NATIVE_BUILD (what we have today)
* COVERAGE_NATIVE_SOURCE is this new LLVM source coverage mode
* COVERAGE_GCOV is the current COVERAGE_GCOV (embedded gcov data generation).

And with COVERAGE_NATIVE_SOURCE being dependent on LLVM (`depends on "${ZEPHYR_TOOLCHAIN_VARIANT}" = "llvm"`) and NATIVE_BUILD.
And with COVERAGE_NATIVE_GCOV  being dependent NATIVE_BUILD.